### PR TITLE
Cleanup mock APIs

### PR DESCRIPTION
**Pull request by andre4i**
## Description

These capabilities have been added as optional in `main` for compatibility reasons. In `next` we can make them required.

Leftover from [AB#4980](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/4980)

### CODE REVIEW

**Review by kian-thompson — APPROVED** [review]

**Comment by tylerbutler** [comment]
I'll merge this after #16611 is merged.

**Pull request merged**
